Unhandled Exceptions: building native assets failed

Hoping someone might be able to help with this. 

```
`Unhandled exception:
Invalid argument(s): Illegal character in path
#0      _Uri._checkWindowsPathReservedCharacters (dart:core/uri.dart:1841:11)
#1      _Uri._makeWindowsFileUrl (dart:core/uri.dart:1929:9)
#2      new _Uri.file (dart:core/uri.dart:1808:11)
#3      FileSystemEntity.uri (dart:io/file_system_entity.dart:244:22)
#4      main.<anonymous closure> (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:187:27)
<asynchronous suspension>
#5      build (package:native_assets_cli/src/api/build.dart:94:3)
<asynchronous suspension>
#6      main (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:8:3)
<asynchronous suspension>
Building native assets for package:thermion_dart failed.
file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart returned with exit code: 255.
To reproduce run:
C:\Users\jackm\flutter\bin\dart --packages=C:\Users\jackm\OneDrive\Documents\GitHub\Simple_Machine_Control\simple_excavator_mc\.dart_tool\package_config.json C:\Users\jackm\AppData\Local\Pub\Cache\hosted\pub.dev\thermion_dart-0.1.1+5\hook\build.dart --config=C:\Users\jackm\OneDrive\Documents\GitHub\Simple_Machine_Control\simple_excavator_mc\.dart_tool\native_assets_builder\c90fbe241c81ff25664205a12660aef7\config.json
stderr:
Unhandled exception:
Invalid argument(s): Illegal character in path
#0      _Uri._checkWindowsPathReservedCharacters (dart:core/uri.dart:1841:11)
#1      _Uri._makeWindowsFileUrl (dart:core/uri.dart:1929:9)
#2      new _Uri.file (dart:core/uri.dart:1808:11)
#3      FileSystemEntity.uri (dart:io/file_system_entity.dart:244:22)
#4      main.<anonymous closure> (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:187:27)
<asynchronous suspension>
#5      build (package:native_assets_cli/src/api/build.dart:94:3)
<asynchronous suspension>
#6      main (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:8:3)
<asynchronous suspension>

stdout:


Target native_assets failed: Error: Building native assets failed. See the logs for more details.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\Users\jackm\flutter\bin\flutter.bat'' finished with non-zero exit value 1` 
```

Happens when i try to build from the example repo as well. 
Long term I need to be able to load and render dxf files on device.  
Thanks!

Thanks mate, 
I tried that. Unfortunately output was basically the same.  
````
PS C:\Users\jackm\Downloads\thermion_examples\flutter\example\lib> flutter run
Changing current working directory to: C:\Users\jackm\Downloads\thermion_examples\flutter\example
Resolving dependencies... (1.1s)
Downloading packages...
  collection 1.18.0 (1.19.0 available)
  flutter_lints 1.0.4 (4.0.0 available)
  image 3.2.2 (4.2.0 available)
  lints 1.0.1 (4.0.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  native_assets_cli 0.6.1 (0.7.0 available)
! native_toolchain_c 0.4.2 from git https://github.com/nmfisher/native.git at 8d11bc in pkgs/native_toolchain_c (overridden)
Got dependencies!
6 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching lib\main.dart on SM T713 in debug mode...
Unhandled exception:
Invalid argument(s): Illegal character in path
#0      _Uri._checkWindowsPathReservedCharacters (dart:core/uri.dart:1841:11)
#1      _Uri._makeWindowsFileUrl (dart:core/uri.dart:1929:9)
#2      new _Uri.file (dart:core/uri.dart:1808:11)
#3      FileSystemEntity.uri (dart:io/file_system_entity.dart:244:22)
#4      main.<anonymous closure> (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:187:27)
<asynchronous suspension>
#5      build (package:native_assets_cli/src/api/build.dart:94:3)
<asynchronous suspension>
#6      main (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:8:3)
<asynchronous suspension>
Building native assets for package:thermion_dart failed.
file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart returned with exit code: 255.
To reproduce run:
C:\Users\jackm\flutter\bin\dart --packages=C:\Users\jackm\Downloads\thermion_examples\flutter\example\.dart_tool\package_config.json C:\Users\jackm\AppData\Local\Pub\Cache\hosted\pub.dev\thermion_dart-0.1.1+5\hook\build.dart --config=C:\Users\jackm\Downloads\thermion_examples\flutter\example\.dart_tool\native_assets_builder\9a643e8009926c9a7e816f0a4b74f07a\config.json
stderr:
Unhandled exception:
Invalid argument(s): Illegal character in path
#0      _Uri._checkWindowsPathReservedCharacters (dart:core/uri.dart:1841:11)
#1      _Uri._makeWindowsFileUrl (dart:core/uri.dart:1929:9)
#2      new _Uri.file (dart:core/uri.dart:1808:11)
#3      FileSystemEntity.uri (dart:io/file_system_entity.dart:244:22)
#4      main.<anonymous closure> (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:187:27)
<asynchronous suspension>
#5      build (package:native_assets_cli/src/api/build.dart:94:3)
<asynchronous suspension>
#6      main (file:///C:/Users/jackm/AppData/Local/Pub/Cache/hosted/pub.dev/thermion_dart-0.1.1+5/hook/build.dart:8:3)
<asynchronous suspension>

stdout:


Target native_assets failed: Error: Building native assets failed. See the logs for more details.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\Users\jackm\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
Running Gradle task 'assembleDebug'...                             13.3s
Error: Gradle task assembleDebug failed with exit code 1
````

--- a/thermion_dart/hook/build.dart
+++ b/thermion_dart/hook/build.dart
@@ -176,9 +176,17 @@ void main(List<String> args) async {
           Architecture.ia32 => "i686-linux-android",
           _ => throw FormatException('Invalid')
         };
-        var ndkRoot = File(config.cCompiler.compiler!.path).parent.parent.path;
+          
+        var compilerPath = config.cCompiler.compiler!.path;
+        
+        if(Platform.isWindows && compilerPath.startsWith("/")) {
+          compilerPath = compilerPath.substring(1);
+        }
+        
+        var ndkRoot = File(compilerPath).parent.parent.uri.toFilePath(windows:true);
+
         var stlPath =
-            File("$ndkRoot/sysroot/usr/lib/${archExtension}/libc++_shared.so");
+            File([ndkRoot, "sysroot", "usr", "lib", archExtension, "libc++_shared.so"].join(Platform.pathSeparator));
         output.addAsset(NativeCodeAsset(
             package: "thermion_dart",
             name: "libc++_shared.so",

